src/collider.h

```diff
@@ -163,14 +163,16 @@ struct FindCollision {
   VecView<const T> queries;
   VecView<const Box> nodeBBox_;
   VecView<const std::pair<int, int>> internalChildren_;
-  Recorder recorder;
+  Recorder& recorder;
 
-  inline int RecordCollision(int node, const int queryIdx, SparseIndices& ind) {
+  using LocalT = typename Recorder::LocalT;
+
+  inline int RecordCollision(int node, const int queryIdx, LocalT& local) {
```

Comment:
For updating other collider usages - I leave it up to you. If it's less work, do it here - otherwise keep them separate for ease of review.